tests/test_odp_zaius_graphql_api_manager.py

```diff
@@ -0,0 +1,364 @@
+                                }
+                            },
+                            {
+                                "node": {
+                                    "name": "b",
+                                    "state": "qualified",
+                                    "description": "qualifed sample 2"
+                                }
+                            },
+                            {
+                                "node": {
+                                    "name": "c",
+                                    "state": "not_qualified",
+                                    "description": "not-qualified sample"
+                                }
+                            }
+                        ]
+                    }
+                }
+            }
+        }
+        """
+
+    good_empty_response_data = """
+        {
+            "data": {
+                "customer": {
+                    "audiences": {
+                        "edges": []
+                    }
+                }
+            }
+        }
+        """
+
+    invalid_identifier_response_data = """
+        {
+          "errors": [
+            {
+              "message": "Exception while fetching data (/customer) :\
+               java.lang.RuntimeException: could not resolve _fs_user_id = asdsdaddddd",
+              "locations": [
+                {
+                  "line": 2,
+                  "column": 3
+                }
+              ],
+              "path": [
+                "customer"
+              ],
+              "extensions": {
+                "classification": "InvalidIdentifierException"
+              }
+            }
+          ],
+          "data": {
+            "customer": null
+          }
+        }
+        """
+
+    other_exception_response_data = """
+        {
+          "errors": [
+            {
+              "message": "Exception while fetching data (/customer) :\
+               java.lang.RuntimeException: could not resolve _fs_user_id = asdsdaddddd",
+              "extensions": {
+                "classification": "TestExceptionClass"
+              }
+            }
+          ],
+          "data": {
+            "customer": null
+          }
+        }
+        """
+
+    bad_response_data = """
+        {
+            "data": {}
+        }
+        """
+    # TODO - for testing key error and log message when getting audiences (dict traversal)
+    invalid_key_response_data = """
+        {
+            "data": {
+                "customer": {
+                    "audiences": {
+                        "invalid_test_key": [
+                            {
+                                "node": {
+                                    "name": "a",
+                                    "state": "qualified",
+                                    "description": "qualifed sample 1"
+                                }
+                            },
```

Comment:
One more invalid data test:
```
                        "edges": [
                            {
                                "node": {
                                    "state": "qualified"
                                }
                            },
                            {
                                "node": {
                                    "name": "a"
                                }
                            },
                            {
                                "other_name": {
                                    "name": "a",
                                    "state": "qualified"
                                }
                            }
```

Reply:
Done. Added tests for node, name, state missing or different, plus the mixed-keys test above.
Good test!